va/va_dec_hevc.h

```diff
@@ -561,6 +565,22 @@ typedef struct  _VASliceParameterBufferHEVCExtension
     VASliceParameterBufferHEVCRext           rext;
 } VASliceParameterBufferHEVCExtension;
 
+/**
+ * \brief HEVC Decoding Subsets Parameter Buffer Structure For Long Format
+ *
+ * The data layout on the slice subsets parameter buffer is concatenation with one or multiple
+ * subset entry point offsets. All the offset values are layed out one by one according to slice
+ * order with first slice segment first, second slice segment second, etc... The total entry number
+ * is indicated by parameter TotalNumEntryPointOffsets(no definition) in VAPictureParameterBufferHEVC.
+ * And the first entry position of the entry point offsets for any slice segment is indicated by
+ * parameter entry_offset_to_subset_array in VAPictureParameterBufferHEVC data structure.
+ */
+typedef struct _VASubsetsParameterBufferHevc
+{
+    /** HEVC syntaxt element. */
+    uint32_t entry_point_offset_minus1[440]; 
+} VASubsetsParameterBufferHevc;
```

Comment:
Could this be defined as an array of uint32_t instead of a structure?  If the buffer is of fixed size like this then it will usually be mostly unused, and may not be big enough for some degenerate streams (with both `tiles_enabled_flag` and `entropy_coding_sync_enabled_flag` the number of entry points can be very large).

Reply:
Should be fine, but I'm not sure what you gain by the extra layer of indirection rather than putting the arbitrary-length array inside the buffer. (If you pass a pointer here then the driver will always have to copy the whole array immediately to some internal location, because the lifetime of an array outside the buffer is not constrained by the API when decoding asynchronously.)

Reply:
@fhvwy For some case where both tile and WPP exist, the number of entry points may be larger than 440. However WPP entry points are not needed for HW decoding, so we only need tile entry points that will not more than 440. As you said, if defining the buffer as fixed size, it will usually be mostly unsued, so it maybe better to allocate an array with the element size pointed by app for entry points